common/src/main/java/org/apache/comet/parquet/Utils.java

```diff
@@ -22,18 +22,20 @@
 import org.apache.parquet.column.ColumnDescriptor;
 import org.apache.parquet.schema.LogicalTypeAnnotation;
 import org.apache.parquet.schema.PrimitiveType;
+import org.apache.spark.sql.internal.SQLConf;
 import org.apache.spark.sql.types.*;
 
+import org.apache.comet.CometConf;
+
 public class Utils {
   public static ColumnReader getColumnReader(
-      DataType type,
-      ColumnDescriptor descriptor,
-      int batchSize,
-      boolean useDecimal128,
-      boolean useLazyMaterialization) {
-    // TODO: support `useLegacyDateTimestamp` for Iceberg
+      DataType type, ColumnDescriptor descriptor, int batchSize, SQLConf conf) {
```

Comment:
I assume we do not need the old signature? It is ok to change now as we have not made a release yet, but just asking because this is a public method.

Reply:
I don't think anybody uses the old signature yet.

Reply:
This will make it harder to make the column reader Spark agnostic as you're adding one more point of  dependency.